framework/MacOnly/CPTPlatformSpecificCategories.m

```diff
@@ -12,30 +12,40 @@ @implementation CPTLayer(CPTPlatformSpecificLayerExtensions)
 -(nonnull CPTNativeImage *)imageOfLayer
 {
     CGSize boundsSize = self.bounds.size;
-
-    NSBitmapImageRep *layerImage = [[NSBitmapImageRep alloc] initWithBitmapDataPlanes:NULL
-                                                                           pixelsWide:(NSInteger)boundsSize.width
-                                                                           pixelsHigh:(NSInteger)boundsSize.height
-                                                                        bitsPerSample:8
-                                                                      samplesPerPixel:4
-                                                                             hasAlpha:YES
-                                                                             isPlanar:NO
-                                                                       colorSpaceName:NSCalibratedRGBColorSpace
-                                                                          bytesPerRow:(NSInteger)boundsSize.width * 4
-                                                                         bitsPerPixel:32];
-
+    
+    // Figure out the scale of pixels to points
+    CGFloat scale = [[NSScreen mainScreen] backingScaleFactor];
```

Comment:
This only considers the main screen, but leaves out multi monitor setups. You should use the scale factor of the monitor the application is actually running on (e.g using the NSWindow.backingScaleFactor).